exposed-tests/src/test/kotlin/org/jetbrains/exposed/sql/tests/shared/entities/CompositeIdTableEntityTest.kt

```diff
@@ -438,4 +439,143 @@ class CompositeIdTableEntityTest : DatabaseTestsBase() {
             assertEqualCollections(publisherA.allOffices.toList(), listOf(officeB))
         }
     }
+
+    // fails because of bug in SQL Server notInList logic - see link to fix in PR details
+//    @Test
+//    fun testInListWithCompositeIdEntities() {
+//        withTables(excludeSettings = listOf(TestDB.SQLITE), Publishers) {
+//            val id1: EntityID<CompositeID> = Publishers.insertAndGetId {
+//                it[name] = "Publisher A"
+//            }
+//            val id2: EntityID<CompositeID> = Publishers.insertAndGetId {
+//                it[name] = "Publisher B"
+//            }
+//
+//            val compositeIds = listOf(id1.value, id2.value)
+//            val keyColumns = Publishers.idColumns.toList()
+//            val result1 = Publishers.selectAll().where { keyColumns inList compositeIds }.count()
+//            assertEquals(2, result1)
+//            val result2 = Publishers.selectAll().where { keyColumns notInList compositeIds }.count()
+//            assertEquals(0, result2)
+//
+//            val result3 = Publishers.selectAll().where { Publishers.id inList compositeIds }.count()
+//            assertEquals(2, result3)
+//            val result4 = Publishers.selectAll().where { Publishers.id notInList compositeIds }.count()
+//            assertEquals(0, result4)
+//        }
+//    }
+
+    @Test
+    fun testPreloadReferencesOnCompositeIdEntities() {
```

Comment:
This test seems to be quite slow, I'm assuming because of the `inTopLevelTransaction`. I will split it locally into multiple tests and compare.